### ISSUE TITLE
DO + ssh = Host key verification failed & API Error

### ISSUE THREAD

**Issue by dfroberg** [supergiant/supergiant#236]
A standard config; 
`"kubernetes_version": "1.5.7",`
With only a fingerprint now results in `Host key verification failed`.
```
{
  "cloud_account_name": "DO Token",
  "digitalocean_config": {
    "region": "nyc1",
    "ssh_key_fingerprint": [
      "f3:77:71:cc:0b:10:1f:a8:bd:55:09:ab:6c:01:d9:bf"
    ]
  },
  "master_node_size": "1gb",
  "name": "amn-kube",
  "node_sizes": [
    "1gb",
  ]
}
```

When attaching a ssh_pub_key AND a fingerprint, the ssh_pub_key is ignored and result is `Host key verification failed`.

```
 {
  "cloud_account_name": "DO Token",
  "digitalocean_config": {
    "region": "nyc1",
    "ssh_key_fingerprint": [
      "f3:77:71:cc:0b:10:1f:a8:bd:55:09:ab:6c:01:d9:bf"
    ]
  },
  "ssh_pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCRSrGw/2bWOBdJo17vSOZ15D+NRKxgDxRd4K+5hqChs5E0NUx1cZf2tpkY/Db/skb3QaIBteYWXZsu6qtxw0tml/5VtXXVtXOF7bF/ZC/xhvKKGMjvDHQvvIX1Gn/DAHV3Hl3RPxjeo3Y/2ObTxMxDnZJiSO/CJJxC0rOkFKBHF7oZzbUWAsxKLxVTAxxp5jEyCrpPACW3K95n4k9d5jbITz3uFWIhXVv9If65uH6a72jyOK93H2Egbu8NlgaF1hZTKeVc9cH1YoanG+qUDd8NKZXd/qQ99gSDoVjR9RGrGVza+2h3wUqsWxfpbwodk0nB0eyr616xI2Vj0QdjNyzN",
  "master_node_size": "1gb",
  "name": "amn-kube",
  "node_sizes": [
    "1gb",
  ]
}
```

Leaving out the fingerprint and only specifying the ssh_pub_key, results in a provisioning error;
`"error": "POST https://api.digitalocean.com/v2/droplets: 422 0 are invalid key identifiers for Droplet creation.",`
```
 {
  "cloud_account_name": "DO Token",
  "digitalocean_config": {
    "region": "nyc1",
    "ssh_key_fingerprint": [
      ""
    ]
  },
  "ssh_pub_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCRSrGw/2bWOBdJo17vSOZ15D+NRKxgDxRd4K+5hqChs5E0NUx1cZf2tpkY/Db/skb3QaIBteYWXZsu6qtxw0tml/5VtXXVtXOF7bF/ZC/xhvKKGMjvDHQvvIX1Gn/DAHV3Hl3RPxjeo3Y/2ObTxMxDnZJiSO/CJJxC0rOkFKBHF7oZzbUWAsxKLxVTAxxp5jEyCrpPACW3K95n4k9d5jbITz3uFWIhXVv9If65uH6a72jyOK93H2Egbu8NlgaF1hZTKeVc9cH1YoanG+qUDd8NKZXd/qQ99gSDoVjR9RGrGVza+2h3wUqsWxfpbwodk0nB0eyr616xI2Vj0QdjNyzN",
  "master_node_size": "1gb",
  "name": "amn-kube",
  "node_sizes": [
    "1gb",
  ]
}
```

**Comment by gopherstein** [comment]
I am currently able to re-create error 3 but not the first two.

**Comment by gopherstein** [comment]
Ok.. so here is the status of things.. 

For problem 1 and 2 check the comma on the node sizes array. 
For problem 3, looks like the fingerprint is simply required. You can't leave it out. However our UI template automatically fills in an array that has an empty string. This causes the verification utility to miss the required field. I am changing it to this. 

```
{
  "cloud_account_name": "",
  "digitalocean_config": {
    "region": "nyc1",
    "ssh_key_fingerprint": []
  },
  "master_node_size": "1gb",
  "name": "",
  "node_sizes": [
    "1gb",
    "2gb",
    "4gb",
    "8gb",
    "16gb",
    "32gb",
    "48gb",
    "64gb"
  ]
}```

**Comment by dfroberg** [comment]
The comma was me cutting down the list size manually, sorry about that.
Ok can confirm that having a single figerprint listed still gives the `Host key verification failed`
Padding with an extra dummy prior to your real key, does work and allows the second key to be used to login.

So;
```
    "ssh_key_fingerprint": [
	  "68:ec:03:11:2c:c9:71:41:fc:91:22:29:85:5d:94:03",
      "f3:77:71:cc:0b:10:1f:a8:bd:55:09:ab:6c:01:d9:bf"
    ]
```
Does NOT work, whereas;
```
    "ssh_key_fingerprint": [
	  "68:ec:03:11:2c:c9:71:41:fc:91:22:29:85:5d:94:03",
      "f3:77:71:cc:0b:10:1f:a8:bd:55:09:ab:6c:01:d9:bf"
    ]
```
Works fine for the second key.

**Comment by gopherstein** [comment]
Re-opening as it seems there may still be an issue. I however am unable to re-create it... Both the examples look the same to me.

**Comment by gopherstein** [comment]
Can you click on your Kube after launch and send what the ssh_key_fingerprint looks like *after* provisioning?

**Comment by gopherstein** [comment]
I am also interested to see... if you spin up an instance with that first key (not using SG) are you able to SSH in?

**Comment by TheKLARKEN** [comment]
Hi, @dfroberg, I'm having trouble reproducing this issue, and I believe it may have been fixed in a previous pull request, as I'm now able to create a DO cluster with a single SSH fingerprint. Can you clone the current master and check whether or not things work now?

Thanks!

**Comment by TheKLARKEN** [comment]
This was closed recently with pull request #419. Please open if anymore issues arise.
